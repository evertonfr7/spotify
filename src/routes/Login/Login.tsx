import Button from '@/components/ui/Button'
import Logo from '@/components/ui/Logo'
import Typography from '@/components/ui/Typography'
import { LOGIN_URL } from '@/constants/spotify'

export function Login(): JSX.Element {
  return (
    <div className="w-full h-screen flex flex-col items-center justify-center gap-4">
      <Logo />
      <Typography>
        Entra com sua conta Spotify clicando no bot√£o abaixo
      </Typography>
      <a href={LOGIN_URL}>
        <Button>Entrar</Button>
      </a>
    </div>
  )
}
