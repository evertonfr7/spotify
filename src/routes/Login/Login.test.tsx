import { render } from '@testing-library/react'
import { Login } from './Login'

describe('Login component', () => {
  test('renders the login route with correct snapshot', () => {
    const tree = render(<Login />)

    expect(tree).toMatchSnapshot()
  })
})
