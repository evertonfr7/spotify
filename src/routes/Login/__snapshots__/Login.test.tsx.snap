// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the login route with correct snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-screen flex flex-col items-center justify-center gap-4"
      >
        <div>
          <img
            alt="Spotify Logo"
            aria-label="Spotify Logo"
            class="w-[114.8px] h-[35px] md:w-[164px] md:h-[50px] block"
            height="50"
            role="img"
            src="[object Object]"
            title="Spotify"
            width="164"
          />
        </div>
        <p
          class="font-default text-[12px]/[18px] font-medium text-white md:text-[14px]/[20px]"
        >
          Entra com sua conta Spotify clicando no botão abaixo
        </p>
        <button
          class="bg-[#57B660] text-[#181414] rounded-3xl px-8 h-[38px] font-default text-[14px]/[18px] font-bold md:h-[42px] md:px-10 md:text-base/[20px] focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]"
          type="button"
        >
          Entrar
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-screen flex flex-col items-center justify-center gap-4"
    >
      <div>
        <img
          alt="Spotify Logo"
          aria-label="Spotify Logo"
          class="w-[114.8px] h-[35px] md:w-[164px] md:h-[50px] block"
          height="50"
          role="img"
          src="[object Object]"
          title="Spotify"
          width="164"
        />
      </div>
      <p
        class="font-default text-[12px]/[18px] font-medium text-white md:text-[14px]/[20px]"
      >
        Entra com sua conta Spotify clicando no botão abaixo
      </p>
      <button
        class="bg-[#57B660] text-[#181414] rounded-3xl px-8 h-[38px] font-default text-[14px]/[18px] font-bold md:h-[42px] md:px-10 md:text-base/[20px] focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-[#57B660]"
        type="button"
      >
        Entrar
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
